### ISSUE TITLE
Python's SCPI rx_arb() is broken on the OS 2.00-18 (beta)

### ISSUE THREAD

**Issue by hadriman** [RedPitaya/RedPitaya#262]
The issue is coming from the device that is now sending "\r\n" after the data and that the  rx_arb() function in redpitaya_scpi.py is not expecting it.

This means that the first call to rx_arb() will work but leave the "\r\n" in the receive buffer and subsequent calls to rx_abs() will fail because of those remaining characters.

**Comment by hadriman** [comment]
Here is my fixed function that takes into account those two extra characters (and I added a check with an assert at the end) : 

``` 
def rx_arb(self):
        """ Recieve binary data from scpi server"""
        numOfBytes = 0
        data=b''
        while len(data) != 1:
            data = self._socket.recv(1)
        if data != b'#':
            return False
        data=b''
        while len(data) != 1:
            data = self._socket.recv(1)
        numOfNumBytes = int(data)
        if numOfNumBytes <= 0:
            return False
        data=b''
        while len(data) != numOfNumBytes:
            data += (self._socket.recv(1))
        #
        numOfBytes = int(data)+2  # <- fix is here
        #
        data=b''
        while len(data) != numOfBytes:
            data += (self._socket.recv(4096))
        assert data[-2:] == b"\r\n"
        #
        return data[:-2] # <- and here
        #
```

**Comment by b-swat** [comment]
Thank you. We're aware of the problem. We already have a solution. You can check it out.

[redpitaya_scpi.zip](https://github.com/RedPitaya/RedPitaya/files/12747675/redpitaya_scpi.zip)

**Comment by hadriman** [comment]
> Thank you. We're aware of the problem. We already have a solution. You can check it out.
> 
> [redpitaya_scpi.zip](https://github.com/RedPitaya/RedPitaya/files/12747675/redpitaya_scpi.zip)

Hello, 

Unfortunately I tried the file you provided and it did not seem to fix my issue...
I believe the modifications in that new file are meant to fix a very related issue but not the one I've experienced.

Your fix tells python to only try to receive the exact amount of data that is expected (without taking into account the two extra bytes from "\r\n" ) instead of always asking for 4096 bytes. 
This means that the two extra bytes will never end up in the "data" bytestring (which is an improvement).
But these two bytes "\r\n" stay in the receive buffer and will still cause subsequent queries to fail (because they rely on the receive buffer to be empty to work properly).

I think for the issue to be completely fixed you would need to add a "self._socket.recv(2)" to flush those two bytes just before returning from the function.
